Library/Homebrew/dependency.rb

```diff
@@ -71,25 +71,30 @@ class << self
     # The default filter, which is applied when a block is not given, omits
     # optionals and recommendeds based on what the dependent has asked for.
     def expand(dependent, deps = dependent.deps, &block)
+      @expand_stack ||= []
+      @expand_stack.push dependent
+
       expanded_deps = []
 
       deps.each do |dep|
-        # FIXME: don't hide cyclic dependencies
         next if dependent.name == dep.name
 
         case action(dependent, dep, &block)
         when :prune
           next
         when :skip
+          next if @expand_stack.include? dep
```

Comment:
This doesn't seem to be right. `@expand_stack` stores `dependent`s which are formula objects. While `dep` is a dependency object.